cmd/dev/app/rule_type/init.go

```diff
@@ -0,0 +1,188 @@
+	dir := "."
+	if len(args) > 0 {
+		dir = args[0]
+	}
+
+	if err := validateRuleTypeName(name); err != nil {
+		return err
+	}
+
+	ruleTypeFileName := filepath.Join(dir, name+".yaml")
+	ruleTypeTestFileName := filepath.Join(dir, name+".test.yaml")
+	ruleTypeTestDataDirName := filepath.Join(dir, name+".testdata")
+
+	if err := assertFilesDontExist(
+		ruleTypeFileName, ruleTypeTestFileName, ruleTypeTestDataDirName); err != nil {
+		return err
+	}
+
+	// Create rule type file
+	if err := createRuleTypeFile(ruleTypeFileName, name); err != nil {
+		return err
+	}
+	cmd.Printf("Created rule type file: %s\n", ruleTypeFileName)
+
+	if !skipTests {
+		// Create rule type test file
+		if err := createRuleTypeTestFile(ruleTypeTestFileName); err != nil {
+			return err
+		}
+		cmd.Printf("Created rule type test file: %s\n", ruleTypeTestFileName)
+
+		// Create rule type test data directory
+		if err := createRuleTypeTestDataDir(ruleTypeTestDataDirName); err != nil {
+			return err
+		}
+		cmd.Printf("Created rule type test data directory: %s\n", ruleTypeTestDataDirName)
+	}
+
+	return nil
+}
+
+func validateRuleTypeName(name string) error {
+	if name == "" {
+		return errors.New("name cannot be empty")
+	}
+
+	validName := regexp.MustCompile(`^[a-zA-Z0-9_]+$`)
+
+	// regexp to validate name
+	if !validName.MatchString(name) {
+		return errors.New("name must only contain alphanumeric characters and underscores")
+	}
+
+	return nil
+}
+
+func assertFilesDontExist(files ...string) error {
+	for _, file := range files {
+		if _, err := os.Stat(file); err == nil {
+
+			return fmt.Errorf("file %s already exists", file)
+		}
+	}
+
+	return nil
+}
+
+func createRuleTypeFile(fileName, name string) error {
+	return createFileWithContent(fileName, fmt.Sprintf(`---
+version: v1
+release_phase: alpha
+type: rule-type
+name: %s
+display_name:  # Display name for the rule type
+short_failure_message:   # Short message to display when the rule fails
+severity:
+  value: medium
+context: {}
+description: |  # Description of the rule type
+guidance: |  # Guidance for the rule type. This helps users understand how to fix the issue.
+def:
+  in_entity: repository  # The entity type the rule applies to
+  rule_schema: {}
+  ingest:
+    type: git
+    git:
+  eval:
+    type: rego
+    rego:
+      type: deny-by-default
+      def: |
+        package minder
+
+        import rego.v1
+
+        default allow := false
+
+        allow if {
+            true
+        }
+        
+        message := "This is a test message"
+`, name))
+}
+
+func createRuleTypeTestFile(fileName string) error {
+	return createFileWithContent(fileName, `---
+tests:
+  - name: "TEST NAME GOES HERE""
+    def: {}
+    params: {}
+    expect: "pass"
+    entity: &test-repo
+      type: repository
+      entity:
+        owner: "coolhead"
+        name: "haze-wave"
+    # http:
+    #   body_file: HTTP_BODY_FILE
+    # git:
+    #   repo_base: REPO_BASE_PATH
```

Comment:
It feels like these could benefit from a bit more documentation in a comment, e.g.

```suggestion
    # When testing a rule, additional content can be supplied
    # from files in the `{{ .RuleName }}.testdata` directory.
    # File paths below are relative to this directory.
    # http:
    #  # Input from the `http` ingest type.
    #  body_file: HTTP_BODY_FILE
    # git:
    #   # Input from the `git` ingest type.  Base paths contain
    #   # directory contents, but do not actually need to be a
    #   # git repository.
    #   repo_base: REPO_BASE_PATH
```